[pytorch][script] op dependency analysis bash driver

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29718 [pytorch][script] op dependency analysis bash driver**
* #29550 [pytorch] add LLVM code analyzer in order to replace static dispatch
* #30558 [pytorch] add irregular c10 op registration/invocation cases to test project

Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

ljk53 committed:
[pytorch][script] op dependency analysis bash driver

Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 scripts/build_code_analyzer.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 scripts/build_code_analyzer.sh
```

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 scripts/build_code_analyzer.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 scripts/build_code_analyzer.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 scripts/build_code_analyzer.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 scripts/build_code_analyzer.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 scripts/build_code_analyzer.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 scripts/build_code_analyzer.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 scripts/build_code_analyzer.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 scripts/build_code_analyzer.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 scripts/build_code_analyzer.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 scripts/build_code_analyzer.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 scripts/build_code_analyzer.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 scripts/build_code_analyzer.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ezyang reviewed (approved):
I approved the original PR, is that right?

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 commented:
> I approved the original PR, is that right?

Not really - anyway I updated the original PR to take a more robust approach (no longer assumes the sequence of instructions in LLVM IR). I will ask people who are familiar with LLVM to help take another look.

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch][script] op dependency analysis bash driver"


Summary:
Move the shell script into this separate PR to make the original PR
smaller and less scary.

Test Plan:
- With stacked PRs:
1. analyze test project and compare with expected results:
```
ANALYZE_TEST=1 CHECK_RESULT=1 tools/code_analyzer/build.sh
```

2. analyze LibTorch:
```
ANALYZE_TORCH=1 tools/code_analyzer/build.sh
```

Differential Revision: [D18474749](https://our.internmc.facebook.com/intern/diff/D18474749)

[ghstack-poisoned]
